src/kinesis.ts

```diff
@@ -1,15 +1,25 @@
 import { Kinesis } from "aws-sdk";
-import { PutRecordsRequestEntry } from "aws-sdk/clients/kinesis";
+import { PutRecordsOutput, PutRecordsRequestEntry } from "aws-sdk/clients/kinesis";
 
-export async function putKinesisRecords(kinesis: Kinesis, streamName: string, logEntries: PublishableStructuredLogData[]): Promise<void> {
+export async function putKinesisRecords(kinesis: Kinesis, streamName: string, logEntries: PublishableStructuredLogData[]): Promise<PutRecordsOutput[]> {
     const records: PutRecordsRequestEntry[] = logEntries.map((entry) => {
         return {
             Data: JSON.stringify(entry),
             PartitionKey: entry.cloudwatchId
         }
     });
-    await kinesis.putRecords({
-        Records: records,
+
+    // Batch records since each put record request supports at most 500 records
+    // https://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecords.html
+    const batchRecords: PutRecordsRequestEntry[][] = [];
+    while (records.length > 0) {
+        batchRecords.push(records.splice(0, 500));
```

Comment:
`splice` makes me a little sad as it's not pure but 🤷‍♂️

Reply:
yeah, I was surprised when I checked that there wasn't a consensus answer on how to batch an array. The other approach in answer to [this](https://stackoverflow.com/questions/7273668/how-to-split-a-long-array-into-smaller-arrays-with-javascript) question doesn't mutate the array (at the cost of slightly more complicated code). Not sure if it's preferable 🤷‍♂️

Reply:
I think it's OK as is. It's an array created inside the function so it's encapsulated.